src/cryptography/fernet.py

```diff
@@ -71,11 +71,14 @@ def _encrypt_from_parts(self, data, current_time, iv):
         return base64.urlsafe_b64encode(basic_parts + hmac)
 
     def decrypt(self, token, ttl=None):
+        timestamp, data = self._get_token_data(token)
+        return self._decrypt_data(data, timestamp, ttl)
+
+    @staticmethod
+    def _get_token_data(token):
```

Comment:
Since this returns a timestamp that may be invalid (we haven't performed the HMAC verification), let's call this `_get_unvalidated_token_data`.

(For our purposes it's okay to not have validated it yet as the call to `_decrypt_data` does the HMAC check)